extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -26,21 +30,51 @@ export class AmpSlideScroll extends BaseCarousel {
     /** @private @const {!Window} */
     this.win_ = this.getWin();
 
+    /** @private @const {!boolean} */
+    this.hasNativeSnapPoints_ = (
+        getStyle(this.element, 'scrollSnapType') != undefined);
     this.element.classList.add('-amp-slidescroll');
 
     /** @private {!Array<!Element>} */
     this.slides_ = this.getRealChildren();
 
+    /** @private {!Element} */
+    this.slidesContainer_ = this.win_.document.createElement('div');
+    this.slidesContainer_.classList.add('-amp-slides-container');
+
+    /** @private {!Array<!Element>} */
+    this.slideWrappers_ = [];
+
     this.slides_.forEach(slide => {
-      const wrapper = this.win_.document.createElement('div');
-      wrapper.appendChild(slide);
-      wrapper.classList.add('-amp-slide-item');
-      this.element.appendChild(wrapper);
+      this.setAsOwner(slide);
+      const slideWrapper = this.win_.document.createElement('div');
+      slideWrapper.appendChild(slide);
+      slideWrapper.classList.add('-amp-slide-item');
+      this.slidesContainer_.appendChild(slideWrapper);
+      this.slideWrappers_.push(slideWrapper);
     });
+    this.element.appendChild(this.slidesContainer_);
+
+    /** @private {number} */
+    this.noOfSlides_ = this.slides_.length;
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    /** @private {number} */
+    this.slideWidth_ = this.getLayoutWidth();
+
+    /** @private {number} */
+    this.previousScrollLeft_ = this.slidesContainer_./*REVIEW*/scrollLeft;
   }
 
   /** @override */
   layoutCallback() {
+    if (this.slideIndex_ == null) {
+      return this.mutateElement(() => {
```

Comment:
No need to use `mutateElement` because it's already in the mutate context + you control all aspects of this mutation and no siblings/children are affected by this.